docs/Using-SRIOV-and-Clear-Containers.md

```diff
@@ -0,0 +1,265 @@
+   sudo update-grub
+   ```
+
+5. Reboot and verify
+
+   Host system should be ready now -- reboot and verify that the expected 
+   cmdline and kernel version is booted (look at `/proc/cmdline` and 
+   `/proc/version`)
+
+   ```
+   sudo reboot
+   ```
+
+6. Verify Intel VT-d has been initialized. Check for the following line in 
+   the `dmesg` output.
+
+   ```
+   DMAR: Intel(R) Virtualization Technology for Directed I/O
+   ```
+
+   On older kernels, the prefix is different.
+   ```
+   PCI-DMA: Intel(R) Virtualization Technology for Directed I/O
+   ```
+
+7. Add `vfio-pci` module
+
+   ```
+   sudo modprobe vfio-pci
+   ```
+
+8. If necessary, add PCI quirk for SR-IOV NIC
+
+   Verify your NIC appears in its own IOMMU group and no other devices 
+   appear in the same group, using the command below:
+   ```
+   $ find /sys/kernel/iommu_groups/ -type l
+   ```
+
+   In the rare case where your PCI NIC does not appear in its own group, 
+   the NIC may not support ACS or you may have chosen to build and run an 
+   old kernel. Depending on your NIC and if it indeed enforces isolation, 
+   you may resolve this by adding a `pcie_acs_override=` option to your kernel 
+   command line and reboot. Detailed information about this option can be found 
+   at [PCIE-ACS-override-option](https://lkml.org/lkml/2013/5/30/513).
+
+## Setting up SR-IOV Device
+
+All the steps in prior sections need to be performed just once to prepare the 
+SR-IOV host systems. The following will be needed per system boot in order to
+facilitate setting up a physical device's virtual functions.
+
+The `sriov_totalvfs` file under `sysfs` for a pci device specifies the total
+number of VFs that can be created. VFs can be created by editing the file `sriov_numvfs`.
+Once VF's are created, you cannot grow or shrink the number of VFs without 
+first setting the number back to zero. Based on this, it is expected that you
+should set the number of VFs for a physical device just once.
+
+1. Add `vfio-pci` device driver
+
+   `vfio-pci` is a driver which is used to reserve a VF PCI device.  Add it:
+   ```
+   $ sudo modprobe vfio-pci
+   ```
+
+2. Find our NICs of interest
+
+   Find PCI details for the NICs in question:
+   ```
+   $ lspci | grep Ethernet
+   00:19.0 Ethernet controller: Intel Corporation Ethernet Connection I217-LM (rev 04)
+   01:00.0 Ethernet controller: Intel Corporation Ethernet Controller 10-Gigabit X540-AT2 (rev 01)
+   01:00.1 Ethernet controller: Intel Corporation Ethernet Controller 10-Gigabit X540-AT2 (rev 01)
+   ```
+   In our case, both 01:00.0 and 01:00.1 are the two ports on our x540-AT2 card 
+   that we'll use.  You can use `lshw` command to get further details on the 
+   controller and verify it indeed supports SR-IOV.
+
+3. Check how many VFs we can create:
+
+   ```
+   $ cat /sys/bus/pci/devices/0000\:01\:00.0/sriov_totalvfs
+   63
+   $ cat /sys/bus/pci/devices/0000\:01\:00.1/sriov_totalvfs
+   63
+   ```
+
+4. Create VFs:
```

Comment:
Lines 192-201 rewrite: 

Create the VFs: 

NOTE: insert command immediately following numbered step

The previous commands create virtual functions by editing `sriov_numvfs`. This example creates one VF per physical device. Note, creating one VF eliminates the usefulness of SR-IOV, and is done for simplicity in this example.